app/services/git/fetches_updated_tracks.rb

```diff
@@ -0,0 +1,44 @@
+class Git::FetchesUpdatedTracks
+
+  QUIET_PERIOD = 30.seconds
+
+  def self.run
+    new.run
+  end
+
+  def self.fetch
+    new.fetch
+  end
+
+  def run
+    loop do
+      fetch
+      sleep QUIET_PERIOD
+    end
+  end
+
+  def fetch
+    find_track_update
+    fetch_track_update if track_update
+  end
+
+  private
+
+  attr_reader :track_update, :track_update_fetch
+
+  def find_track_update
+   @track_update = TrackUpdate.
+     includes(:track_update_fetches).
+     where(synced_at: nil).
+     find_by(track_update_fetches: { track_update_id: nil }) ||
```

Comment:
I find your code very readable @kntsoriano. This method did confuse me a bit though, wonder if the `||` here could be made a little clearer, I didn't spot it at first.

Reply:
Sure, I'll come up with a way.